File: python/p010.py
"""
    
Project Euler - Problem 10 - Summation of primes
https://projecteuler.net/problem=10

"""
import timing
from eulerutil import primes

def main():
	
	answer = 0
	for p in primes(2000000):
		answer += p

	print(answer)


if __name__ == '__main__':
    main()

File: js/p010.js
"use strict";
const primes = require("./util/tools.js").sieveEratosthenes;
/*
 * Summation of primes
 * Problem 10
 * The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
 * 
 * Find the sum of all the primes below two million.
 * 
 * Answer: 142913828922
 */

function init() {
    var a = 0,
    	i = 0,
    	p = primes(2000000),
    	l = p.length;

    for (i=0; i<l; i++) 
    {
        a += p[i];
    }

    return a;
}

module.exports.init = init;
